crates/rustc_codegen_spirv/src/builder/mod.rs

```diff
@@ -134,42 +155,58 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             pointee: result_pointee_type,
         }
         .def(self.span(), self);
-        if self.builder.lookup_const_u64(indices[0]) == Some(0) {
+
+        let ptr_id = ptr.def(self);
+        if let Some((original_ptr, mut original_indices)) = self.find_access_chain(ptr_id) {
+            // Transform the following:
+            // OpAccessChain original_ptr [a, b, c]
+            // OpPtrAccessChain ptr base [d, e, f]
+            // into
+            // OpAccessChain original_ptr [a, b, c + base, d, e, f]
+            // to remove the need for OpPtrAccessChain
+            let last = original_indices.last_mut().unwrap();
+            *last = self
+                .add(last.with_type(indices[0].ty), indices[0])
+                .def(self);
```

Comment:
Just looked at the CI failure, this doesn't do constant-folding, you may need to change this other thing:
https://github.com/EmbarkStudios/rust-gpu/blob/3067848d1c7df819a0036eff4ee1ad60a4a3bc0b/crates/rustc_codegen_spirv/src/builder/builder_methods.rs#L644-L657

Something like this:
```rust
simple_op! {
    add, i_add,
    fold_const { 
        int(a, b) => a + b
    }
}
```

However, I think the existing one should be `.wrapping_mul` and this one should be `.wrapping_add`, not `*` and `+` (either that or [the `fold_const` handling](https://github.com/EmbarkStudios/rust-gpu/blob/3067848d1c7df819a0036eff4ee1ad60a4a3bc0b/crates/rustc_codegen_spirv/src/builder/builder_methods.rs#L51-L55) should be using `Wrapping` around the `u128` values).

Reply:
Ah yep! Was curious about what rust's behavior here is, looks like there's a deny-by-default warning ([that people want to make into an error](https://github.com/rust-lang/rust/issues/71800), but looks like that missed the 2021 train) for constant overflow - mmmaybe we want to also error instead of overflow, but who cares right now tbh :P